php/libraries/TimePoint.class.inc

```diff
@@ -1212,7 +1218,7 @@ class TimePoint implements \LORIS\StudyEntities\AccessibleResource
     {
         $centers = $user->getCenterIDs();
 
-        $projMatch   = in_array($this->getProjectID(), $user->getProjectIDs(), true);
```

Comment:
Loose checking can return weird results.

Reply:
The PHP documentation for `in_array` is incorrect (according to my testing..) From what I can tell when the third parameter is true what it actually does is verifies with a `===` check instead of `==`. This results in the `in_array` requiring it to be the exact same instance of a class when the array is an array of objects (rather than simply checking the type (as the documentation says). Removing the strict checking still enforces types (since it's an object the object won't be `==` to other types either.) The note in the documentation about third parameter being to "validate types" seems to only be for built in types that get compared loosely like string/int/etc.

Reply:
Makes sense. Thanks for the clarification.